bots/make-checkout

```diff
@@ -52,24 +53,31 @@ def main():
             sys.stderr.write("> " + output + "\n")
         return output
 
-    execute("git", "fetch", "origin", opts.ref)
+    if opts.repo:
+        if "test\n" in subprocess.check_output([ "git", "remote" ], universal_newlines=True):
+            execute("git", "remote", "remove", "test")
+        execute("git", "remote", "add", "test", "https://github.com/" + opts.repo)
+
+    execute("git", "fetch", "test" if opts.repo else "origin", opts.ref)
     execute("git", "checkout", "--detach", opts.revision)
```

Comment:
So the idea is that the revision argument includes the `test/` remote name prefix now? This name is an internal implementation detail, so should perhaps not be exposed to callers? I. e. use `"test/" + opts.revision?

Reply:
`opts.revision` is a SHA1, no?  It doesn't need to be qualified by a remote.  `opts.ref` is the symbolic name, such as `pull/2/head`.

I am not super sure why we need both, actually.  Git doesn't allow fetching a raw SHA1, but we also don't want to pick up the wrong SHA1 should the ref have changed in between.  So we fetch the ref and check out the SHA1.

Reply:
Ah, thanks. I thought that revision could be any tree-ish, but if we just use it with SHAs, then it's ok. I initially thought the ref would still be valid after rebasing, but they usually aren't, so I'm also not sure why both are passed.